src/data-expander.ts

```diff
@@ -86,13 +86,20 @@ export function complexObjects (gitlabData: any) {
 }
 
 export function needsComplex (data: any) {
-    return {
+    const needs: Need = {
         job: data.job ?? data,
         artifacts: data.artifacts ?? true,
-        optional: data.optional ?? false,
         ...(data.pipeline ? {pipeline: data.pipeline} : {}),
         ...(data.project ? {project: data.project} : {}),
+        ...(data.ref ? {ref: data.ref} : {}),
+        ...(data.optional ? {optional: data.optional} : {}),
     };
+
+    // In needs:project/needs:pipeline, `optional` is not an allowed property
```

Comment:
![image](https://github.com/firecow/gitlab-ci-local/assets/28011783/dd3fb4f5-925c-41b7-ab68-1a8ab587377c)

Reply:
The json schema is quite complex, yeah :smile: